include/envoy/singleton/manager.h

```diff
@@ -0,0 +1,74 @@
+#pragma once
+
+#include <memory>
+
+#include "envoy/common/pure.h"
+#include "envoy/singleton/instance.h"
+
+namespace Envoy {
+namespace Singleton {
+
+/**
+ * An abstract registration for a singleton entry.
+ */
+class Registration {
+public:
+  virtual ~Registration() {}
+  virtual std::string name() PURE;
+};
+
+/**
+ * A concrete implementation of a singleton registration. All singletons are referenced by name
+ * and must be statically registered ahead of time. This can be done like so:
+ *
+ * static constexpr char foo_singleton_name[] = "foo_singleton";
+ * static Registry::RegisterFactory<Singleton::RegistrationImpl<foo_singleton_name>,
```

Comment:
I did this because it's less typing (technically it is a concrete type, it's just an instantiation of RegistrationImpl). We only need name() which is required by registry. I could probably modify Registry to support a name only registration but that seemed like overkill. If you feel otherwise I can do that.